src/wayland-egldisplay.c

```diff
@@ -435,8 +436,9 @@ registry_handle_global(void *data,
         /*
          * Version 3 added format modifier support, which the dmabuf
          * support in this library relies on.
+         * If the version is >= 4 we already bound it in getServerProtocolsInfo
```

Comment:
I think it might simplify thing a little bit to always bind `zwp_linux_dmabuf_v1` during `eglGetPlatformDisplay`, regardless of whether version 3 or 4 is supported. Of course, if version 4 isn't supported we also need to bind `wl_drm`. Then we can get rid of the code that binds it during `eglInitialize`.

Reply:
Sounds good. This sadly still leaves on issue:
if the display is initialized and then destroyed we also free all wayland objects in terminateDisplay. The spec allows initializing the same display again so we would still need to bind the interfaces in eglInitialize if they aren't bound already.

Reply:
Ah shit, true, I had forgotten about that. Well, if we might need to re-bind the interface during `eglInitialize` anyway maybe it would be better to just go with the original version of your patch. Now I feel like a bit of an ass for making you re-write it, really sorry about that.

Reply:
No problem. I initially also forgot about that case.
I reverted to the initial patch but kept the new error handling in getServerProtocolsInfo